src/syntax/ast.rs

```diff
@@ -82,6 +82,42 @@ pub struct StmtAssign {
     pub value: Option<ExprS>,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct StmtFieldAssign {
+    pub access_modifier: Option<AccessModifier>,
+    pub field: Spanned<StmtAssign>,
+}
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum AccessModifier {
+    Invalid,
+    Private,
+    Public,
+}
+
+impl From<&str> for AccessModifier {
+    fn from(value: &str) -> Self {
+        let access_modifier = match value {
+            "public" => AccessModifier::Public,
+            "private" => AccessModifier::Private,
+            _ => AccessModifier::Invalid,
```

Comment:
`AccessModifier::Invalid` and this `From` implementation are both odd and might go away.